neurodsp/tests/test_timefrequency_hilbert.py

```diff
@@ -1,42 +1,83 @@
 """Test functions for time-frequency Hilbert analyses."""
 
-from neurodsp.tests.settings import FS
+import numpy as np
+
+from neurodsp.tests.settings import FS, N_SECONDS, EPS
 
 from neurodsp.timefrequency.hilbert import *
 
+from neurodsp.utils.data import create_times
+
 ###################################################################################################
 ###################################################################################################
 
-def test_robust_hilbert():
+def test_robust_hilbert(tsig_sine):
 
     # Generate a signal with NaNs
     fs, n_points, n_nans = 100, 1000, 10
     sig = np.random.randn(n_points)
     sig[0:n_nans] = np.nan
 
     # Check has correct number of nans (not all nan), without increase_n
-    hilb_sig = robust_hilbert(sig)
+    hilb_sig = robust_hilbert(sig, False)
     assert sum(np.isnan(hilb_sig)) == n_nans
 
     # Check has correct number of nans (not all nan), with increase_n
     hilb_sig = robust_hilbert(sig, True)
     assert sum(np.isnan(hilb_sig)) == n_nans
 
-def test_phase_by_time(tsig):
+    # Hilbert transform of sin(omega * t) = -sin(omega) * cos(omega * t)
+    times = create_times(N_SECONDS, FS)
+
+    # omega = 1
+    hilbert_sig = np.imag(robust_hilbert(tsig_sine))
+    answer = np.array([-np.cos(2*np.pi*time) for time in times])
+    assert np.max(abs(hilbert_sig-answer)) < EPS
+
+    # omega = -1
+    hilbert_sig = np.imag(robust_hilbert(-tsig_sine))
+    answer = np.array([np.cos(2*np.pi*time) for time in times])
+    assert np.max(abs(hilbert_sig-answer)) < EPS
+
+def test_phase_by_time(tsig, tsig_sine):
 
+    # Check that a random signal, with a filter applied, runs & preserves shape
     out = phase_by_time(tsig, FS, (8, 12))
     assert out.shape == tsig.shape
 
-def test_amp_by_time(tsig):
+    # Check the expected answer for the test sine wave signal
+    #   The instantaneous phase of sin(t) should be piecewise linear with slope 1
+    phase = phase_by_time(tsig_sine, FS)
 
+    # Create a time axis, scaled to the range of [0, 2pi]
+    times = 2 * np.pi * create_times(N_SECONDS, FS)
+    # Generate the expected instantaneous phase of the given signal
+    answer = np.array([time-np.pi/2 if time <= 3*np.pi/2 else time-5*np.pi/2 for time in times])
```

Comment:
This line is a bit dense - and to me is a bit of a mystery (at least, takes some time to parse). Could you add some description to give a sense of what this is?

I also think we should consolidate on using the name `expected_answer` or maybe `exp_answer`. Maybe it's just me, but I read `answer` as "proposed answer" (as in, the function return) and not necessarily to answer to check against.

Reply:
I agree this seems unusual at first. Sin has a period of 2pi seconds, so it's easier to know what the phase should be at time t in [0, 2pi] rather than in [0,1]. The operation `time-np.pi/2 if time <= 3*np.pi/2 else time-5*np.pi/2` in the list comprehension adjusts for the fact that phase is defined to be in [-pi, pi] rather than [0, 2pi].

Reply:
Ahh, I see. So, what about adding a `create_radians` function, to add to `utils`, to work a bit like `create_times` and address this use case. This would simplify here, but might be somewhat generally useful to add to the module. 

API would probably be:
`def create_times(n_radians, fs, start_val=0)`

Which could be used in this test as :
`rads = create_times(2*np.pi, fs, start_val=-np.pi)`